frame/clearing-house/src/tests/update_funding.rs

```diff
@@ -38,6 +41,39 @@ fn get_position(account: &AccountId, market_id: &MarketId) -> Position {
 //                                           Unit tests
 // -------------------------------------------------------------------------------------------------
 
+#[test]
+fn should_update_oracle_and_vamm_twaps() {
+	let config = MarketConfig {
+		funding_frequency: ONE_HOUR,
+		funding_period: ONE_HOUR,
+		twap_period: ONE_HOUR,
+		..Default::default()
+	};
+
+	// Set oracle price at genesis
+	let ext_builder = ExtBuilder { oracle_price: Some(100) /* 1.0 */, ..Default::default() };
+	with_market_context(ext_builder, config, |market_id| {
+		// Get variables before update (timestamp should be 0)
+		let Market { vamm_id, last_oracle_price, last_oracle_twap, .. } = get_market(&market_id);
+		assert_eq!(last_oracle_price, 0.into());
+		assert_eq!(last_oracle_twap, 0.into());
+
+		// Run to next available update time
+		run_to_time(ONE_HOUR);
+		set_oracle_price(&market_id, 1.into());
+		VammPallet::set_next_twap_of(&vamm_id, Some(1.into()));
+		assert_ok!(<TestPallet as ClearingHouse>::update_funding(&market_id));
+
+		let market = get_market(&market_id);
+		// First update since market creation, so TWAP should be the same as current price
+		assert_eq!(market.last_oracle_price, 1.into());
+		assert_eq!(market.last_oracle_twap, market.last_oracle_price);
+		assert_eq!(market.last_oracle_ts, ONE_HOUR);
+		// Assert `Vamm::update_twap` was called
+		assert_eq!(VammPallet::get_twap(vamm_id, AssetType::Base).unwrap(), 1.into());
```

Comment:
The current implementation of `get_twap` for the Vamm simply return the twap stored in the runtime, without worrying to update it, which would require a separate call to `update_twap`. (twap updates happen automatically during swaps though). 

Would it be desirable to try to update the twap every time a call to `get_twap` happens?

Reply:
I'm testing that `update_funding` calls `update_twap` on the Vamm internally. If you see how I implemented `set_next_twap_of` in the mock Vamm, it's supposed to be followed by a call to `update_twap`.

Side comment: `update_funding` is the only place I call `update_twap` on the Vamm explicitly. For `open_position` and `close_position`, I count on the Vamm doing that automatically before each swap